src/Form/atoms/FilterOption.tsx

```diff
@@ -0,0 +1,178 @@
+      fill: ${({ theme }) => theme.colors.icons.inverse};
+    }
+  }
+`;
+
+const FakeRadioButton = styled.div`
+  position: relative;
+  display: inline-block;
+  width: 18px;
+  height: 18px;
+  user-select: none;
+  border-radius: 50%;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  border-width: 2px;
+  border-style: solid;
+`;
+
+const FakeInnerRadio = styled.div`
+  width: 10px;
+  height: 10px;
+  border-radius: 50%;
+  user-select: none;
+`;
+
+const Container = styled.div<{ disabled: boolean, selected: boolean }>`
+  display: flex;
+  align-items: center;
+  height: 100%;
+  width: 100%;
+
+  ${({ theme: { styles }, selected, disabled }) => styles && css`
+    ${FakeCheckbox}, ${FakeRadioButton} {
+      ${styles.form.checkbox.unchecked.default};
+    }
+
+    &:hover {
+      cursor: pointer;
+      ${Title} {
+        color: ${styles.form.checkbox.checked.hover.backgroundColor};
+      }
+
+      ${FakeCheckbox} {
+        ${styles.form.checkbox.unchecked.hover};
+      }
+
+      ${FakeRadioButton} {
+        border-color: ${styles.form.checkbox.unchecked.hover.borderColor};
+      }
+    }
+
+    ${selected && css`
+      ${Title} {
+        color: ${styles.form.checkbox.checked.default.backgroundColor};
+      }
+      ${FakeCheckbox} {
+        ${styles.form.checkbox.checked.default};
+        border: none;
+      }
+
+      ${FakeRadioButton} {
+        border-color: ${styles.form.checkbox.checked.default.backgroundColor};
+        ${FakeInnerRadio} {
+          background-color: ${styles.form.checkbox.checked.default.backgroundColor};
+        }
+      }
+
+      &:hover {
+        ${FakeCheckbox}{
+          ${styles.form.checkbox.checked.hover};
+        }
+
+        ${FakeRadioButton} {
+          border-color: ${styles.form.checkbox.checked.hover.backgroundColor};
+          ${FakeInnerRadio} {
+            background-color: ${styles.form.checkbox.checked.hover.backgroundColor};
+          }
+        }
+      }
+    `};
+
+    ${disabled && css`
+      cursor: not-allowed;
+    `};
+  `};
+`;
+
+interface IFilterOption {
+  title: string
+  value: string | number
+  optionType?: IInputOptionsType
+  selected?: boolean
+  disabled?: boolean
+  onClick?: () => void
+}
+
+const FilterOption: React.FC<IFilterOption> = ({
+  title,
+  value,
+  optionType = "text",
+  selected = false,
+  disabled = false,
+  onClick = () => { },
+  ...props
+}) => {
+
+  const iconWeight: number = dimensions.icons.weights['regular'];
+
+  return (
+
+    <Container
+      {...{ onClick, disabled, selected}}
+      {...props}
+    >
+      {(optionType === 'checkbox') && (
+        <FakeCheckbox>
+          <FakeCheckboxInner>
+            {selected && <CheckMarkWrapper><CheckMark color='inverse' stroke='inverse' size={12} weight={iconWeight} /></CheckMarkWrapper>}
+          </FakeCheckboxInner>
+        </FakeCheckbox>
+      )}
+      {(optionType === 'radio') && <FakeRadioButton><FakeInnerRadio /></FakeRadioButton>}
```

Comment:
Why are we using fake checkboxes and radio buttons instead of real ones?

Reply:
We talk about this over a call.

Summary:
My point of view is that the Hover state requires to have the parent controlling the child input. The extra code is not adding value to either component just for the UI visual. The current solution covers the requirements.

Leonard will review other possibilities that might be updated in another PR.